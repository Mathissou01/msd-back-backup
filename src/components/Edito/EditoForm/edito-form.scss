@use "src/styles/01-settings/settings.colors" as colors;
@use "src/styles/02-tools/tools.shadows" as shadows;

.c-EditoForm {
  width: 100%;

  &__Buttons {
    display: flex;
    gap: 16px;
    justify-content: right;
    height: 48px;
    margin-bottom: 55px;
  }

  &__Form {
    display: flex;
    flex-direction: row;
    gap: 16px;
    width: 100%;
  }

  &__Content {
    display: flex;
    flex-direction: column;
    gap: 32px;
    width: 77%;
  }

  &__SideBar {
    width: 23%;
  }
}
