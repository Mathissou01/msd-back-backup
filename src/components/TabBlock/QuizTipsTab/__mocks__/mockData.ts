import { GetQuizAndTipsBlockDocument } from "../../../../graphql/codegen/generated-types";

export const defaultMockData = [
  {
    request: {
      query: GetQuizAndTipsBlockDocument,
      variables: {
        contractId: "1",
      },
    },
    result: {
      data: {
        contractCustomizations: {
          data: [
            {
              attributes: {
                homepage: {
                  data: {
                    attributes: {
                      quizAndTipsBlock: {
                        data: {
                          id: "1",
                          attributes: {
                            title: "Titre modifi√©",
                            displayBlock: false,
                            displayQuiz: false,
                            quiz: {
                              data: {
                                id: "2",
                                attributes: {
                                  title: "Nom du quiz",
                                  publishedAt: "2022-11-20T02:34:14.535Z",
                                },
                              },
                            },
                            displayTips: false,
                            tips: {
                              data: [],
                            },
                          },
                        },
                      },
                    },
                  },
                },
              },
            },
          ],
        },
        services: {
          data: [
            {
              attributes: {
                serviceInstance: [
                  {
                    __typename: "ComponentMsdEditorial",
                    id: "4",
                    editorialServices: {
                      data: [
                        {
                          attributes: {
                            subServiceInstance: [
                              {
                                __typename: "ComponentEditoQuizzesSubService",
                                isActivated: true,
                                quizzes: {
                                  data: [
                                    {
                                      id: "2",
                                      attributes: {
                                        title: "Nom du quiz",
                                        publishedAt: "2022-11-20T02:34:14.535Z",
                                      },
                                    },
                                    {
                                      id: "1",
                                      attributes: {
                                        title: "truc",
                                        publishedAt: "2022-11-18T10:16:08.936Z",
                                      },
                                    },
                                  ],
                                },
                              },
                            ],
                          },
                        },
                        {
                          attributes: {
                            subServiceInstance: [
                              {
                                __typename: "ComponentEditoTipsSubService",
                                isActivated: true,
                                tips: {
                                  data: [
                                    {
                                      id: "5",
                                      attributes: {
                                        title: "nah",
                                        publishedAt: "2022-11-19T00:14:37.403Z",
                                      },
                                    },
                                    {
                                      id: "2",
                                      attributes: {
                                        title: "Titre de l'astuce 1",
                                        publishedAt: "2022-11-19T00:14:08.232Z",
                                      },
                                    },
                                    {
                                      id: "1",
                                      attributes: {
                                        title: "Titre de l'astuce 2",
                                        publishedAt: "2022-11-18T10:17:11.158Z",
                                      },
                                    },
                                    {
                                      id: "3",
                                      attributes: {
                                        title: "Titre de l'astuce 3",
                                        publishedAt: "2022-11-19T00:14:20.490Z",
                                      },
                                    },
                                    {
                                      id: "4",
                                      attributes: {
                                        title: "Titre de l'astuce 4",
                                        publishedAt: "2022-11-19T00:14:30.273Z",
                                      },
                                    },
                                  ],
                                },
                              },
                            ],
                          },
                        },
                      ],
                    },
                  },
                ],
              },
            },
          ],
        },
      },
    },
  },
];

export const errorMockData = [
  {
    request: {
      query: GetQuizAndTipsBlockDocument,
      variables: {
        contractId: "1",
      },
    },
    error: new Error("An error occured"),
  },
];
