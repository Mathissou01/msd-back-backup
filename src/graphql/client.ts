import {
  ApolloClient,
  InMemoryCache,
  NormalizedCacheObject,
} from "@apollo/client";

let client: ApolloClient<NormalizedCacheObject>;

if (process.env.NEXT_PUBLIC_MOCK === "true") {
  /* With .env variable NEXT_PUBLIC_MOCK=true, return mock graphql client with fake response data */
  client = (await import("./mockedClient")).default(1000);
} else {
  client = await new ApolloClient({
    uri: `${process.env.NEXT_PUBLIC_API_URL}/graphql`,
    cache: new InMemoryCache({
      typePolicies: {
        UploadFolder: {
          merge: true,
        },
        UploadFile: {
          merge: true,
        },
        Contract: {
          merge: true,
        },
        ContractCustomization: {
          merge: true,
        },
        New: {
          merge: true,
        },
        Tip: {
          merge: true,
        },
        RecyclingGuideService: {
          merge: true,
        },
        WasteForm: {
          merge: true,
        },
        WasteFamily: {
          merge: true,
        },
        Flow: {
          merge: true,
        },
        FlowColor: {
          merge: true,
        },
        DropOffMap: {
          merge: true,
        },
        Request: {
          merge: true,
        },
        RequestAggregate: {
          merge: true,
        },
      },
    }),
  });
}

export default client;
