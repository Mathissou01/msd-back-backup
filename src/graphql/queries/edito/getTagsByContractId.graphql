query getTagsByContractId($contractId: ID) {
  tags(filters: { contract: { id: { eq: $contractId } } }) {
    data {
      id
      attributes {
        name
      }
    }
  }
}
