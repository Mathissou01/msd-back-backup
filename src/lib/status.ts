export enum EStatus {
  Archived = "archived",
  Draft = "draft",
  Published = "published",
  Activated = "activated",
}

export enum EStatusLabel {
  draft = "Brouillon",
  published = "Publié",
  archived = "Archivé",
  activated = "Actif",
}

const statusMap = new Map<string, EStatus>(
  Object.values(EStatus).map((v) => [v, v]),
);

export function valueToEStatus(value?: string | null): EStatus {
  const mapped = value ? statusMap.get(value) : EStatus.Draft;
  return mapped ?? EStatus.Draft;
}

export const statusLabels = {
  archived: "Archivé",
  draft: "Brouillon",
  published: "Publié",
  activated: "Actif",
};
