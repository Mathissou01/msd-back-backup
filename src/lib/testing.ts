import { prettyDOM } from "@testing-library/react";

export function removeUnstableHtmlPropertiesForTesting(
  htmlElement: HTMLElement,
  attributeName: string,
) {
  const domHTML = prettyDOM(htmlElement, Infinity).toString();
  if (!domHTML) return undefined;
  const regex = new RegExp(`${attributeName}(.*)"(.*)"`, "g");
  return domHTML.replaceAll(regex, "");
}
