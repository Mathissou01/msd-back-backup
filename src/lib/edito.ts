import { Maybe } from "../graphql/codegen/generated-types";
import { TPictoStyles } from "./pictos";
import { removeNulls } from "./utilities";
import { EStatus } from "./status";
import { ICommonSelectOption } from "../components/Form/FormSingleMultiselect/FormSingleMultiselect";

export interface IEditoStaticFields {
  id: string;
  status: EStatus;
  title: string;
  tags?: Array<ICommonSelectOption>;
  shortDescription?: Maybe<string>;
  publishedDate?: Date;
  unpublishedDate?: Date;
}

export interface IEditoFields extends IEditoStaticFields {
  blocks: Array<IEditoBlock>;
}

/* Blocks */
export type TBlocksDynamicZone =
  | "ComponentBlocksFile"
  | "ComponentBlocksHorizontalRule"
  | "ComponentBlocksImage"
  | "ComponentBlocksSubHeading"
  | "ComponentBlocksVideo"
  | "ComponentBlocksWysiwyg"
  | "Error";
export type TDynamicFieldOption = Exclude<TBlocksDynamicZone, "Error">;

interface IBlockDisplayMap {
  label: string;
  picto: TPictoStyles;
  isEmpty?: boolean;
}

export const blockDisplayMap: Record<TDynamicFieldOption, IBlockDisplayMap> = {
  ComponentBlocksFile: {
    label: "Fichier",
    picto: "attachment",
  },
  ComponentBlocksHorizontalRule: {
    label: "Séparateur",
    picto: "expandVertical",
    isEmpty: true,
  },
  ComponentBlocksImage: {
    label: "Image",
    picto: "picture",
  },
  ComponentBlocksSubHeading: {
    label: "Titre",
    picto: "text",
  },
  ComponentBlocksVideo: {
    label: "Vidéo",
    picto: "video",
  },
  ComponentBlocksWysiwyg: {
    label: "Texte",
    picto: "text",
  },
};

interface IPartialBlockDynamicZone {
  __typename: TBlocksDynamicZone;
  id: string;
}

interface IPartialBlock {
  __typename: TDynamicFieldOption;
  id: string;
}

export type IEditoBlock =
  | IPartialBlock
  | IBlocksFile
  | IBlocksHorizontalRule
  | IBlocksImage
  | IBlocksSubHeading
  | IBlocksVideo
  | IBlocksWysiwyg;

export interface IBlocksFile extends IPartialBlock {
  __typename: "ComponentBlocksFile";
}

export interface IBlocksHorizontalRule extends IPartialBlock {
  __typename: "ComponentBlocksHorizontalRule";
  hr: string;
}

export interface IBlocksImage extends IPartialBlock {
  __typename: "ComponentBlocksImage";
}

export interface IBlocksSubHeading extends IPartialBlock {
  __typename: "ComponentBlocksSubHeading";
  subHeadingTag?: string;
  subHeadingText?: string;
}

export interface IBlocksVideo extends IPartialBlock {
  __typename: "ComponentBlocksVideo";
  videoLink?: string;
  transcriptText?: string;
}

export interface IBlocksWysiwyg extends IPartialBlock {
  __typename: "ComponentBlocksWysiwyg";
}

/* Methods */
export function isEditoBlock(
  block: Partial<IPartialBlockDynamicZone>,
): block is IEditoBlock {
  return "__typename" in block && "id" in block;
}

export function remapEditoBlocksDynamicZone(
  blocks?: Array<Partial<IPartialBlockDynamicZone> | null> | null,
): Array<IEditoBlock> {
  return (
    blocks
      ?.map((block) => {
        if (block) {
          const type = block.__typename;
          if (type && type !== "Error" && isEditoBlock(block)) {
            return {
              ...block,
              __typename: block.__typename,
              id: block.id,
            };
          }
        }
      })
      .filter(removeNulls) ?? []
  );
}

export function createEmptyBlock(
  __typename: TDynamicFieldOption,
  id: string,
): IEditoBlock {
  switch (__typename) {
    case "ComponentBlocksHorizontalRule": {
      return {
        __typename,
        id,
        hr: "<hr>",
      };
    }
    case "ComponentBlocksSubHeading": {
      return {
        __typename,
        id,
        subHeadingText: undefined,
        subHeadingTag: undefined,
      };
    }
    case "ComponentBlocksVideo": {
      return {
        __typename,
        id,
        videoLink: undefined,
        transcriptText: undefined,
      };
    }
    default: {
      return { __typename, id };
    }
  }
}
