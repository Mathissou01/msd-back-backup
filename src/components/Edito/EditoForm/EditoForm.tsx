import _ from "lodash";
import { FieldValues } from "react-hook-form/dist/types/fields";
import { FormProvider, useForm } from "react-hook-form";
import { useEffect, useState } from "react";
import { removeNulls } from "../../../lib/utilities";
import { IEditoFields, TDynamicFieldOption } from "../../../lib/edito";
import EditoButtons from "./EditoButtons/EditoButtons";
import EditoStaticFields from "./EditoStaticFields/EditoStaticFields";
import EditoDynamicFields from "./EditoDynamicFields/EditoDynamicFields";
import EditoSideBar from "./EditoSideBar/EditoSideBar";
import "./edito-form.scss";

interface IEditoFormProps {
  data?: IEditoFields;
  dynamicFieldsOptions: Array<TDynamicFieldOption>;
  onSubmitValid: (data: FieldValues) => void;
}

export default function EditoForm({
  data,
  dynamicFieldsOptions,
  onSubmitValid,
}: IEditoFormProps) {
  /* Static data */

  /* Local Data */
  const form = useForm({
    mode: "onChange",
    defaultValues: data,
    shouldFocusError: false,
  });
  const [formData, setFormData] = useState<IEditoFields>();
  const [canFocus, setCanFocus] = useState(false);
  const { handleSubmit } = form;

  function onError() {
    setCanFocus(true);
  }

  useEffect(() => {
    if (form.formState.errors && canFocus) {
      const errors = form.formState.errors;
      const elements = Object.keys(errors)
        .map((name) => {
          if (name === "blocks" && errors[name]) {
            const blocksErrors = { ...errors[name] };
            return Object.keys(blocksErrors)
              .map((block) => {
                const blockErrors = _.get(blocksErrors, block);
                return Object.keys(blockErrors).map((field) =>
                  document.getElementById(`${name}.${block}.${field}`),
                );
              })
              .flat();
          }
          return document.getElementById(name);
        })
        .flat()
        .filter(removeNulls);
      elements.sort(
        (a, b) => a.getBoundingClientRect().top - b.getBoundingClientRect().top,
      );
      if (elements.length > 0) {
        const errorElement = elements[0];
        setTimeout(() => {
          errorElement.focus({ preventScroll: true });
          errorElement.scrollIntoView({ behavior: "smooth", block: "center" });
        }, 100);
        setCanFocus(false);
      }
    }
  }, [form, canFocus]);

  useEffect(() => {
    if (data) {
      const mappedData = {
        ...data,
      };
      setFormData(mappedData);
      form.reset(mappedData);
    }
  }, [data, form]);

  return (
    <>
      <FormProvider {...form}>
        <form
          className="c-EditoForm"
          onSubmit={handleSubmit(onSubmitValid, onError)}
        >
          <div className="c-EditoForm__Buttons">
            <EditoButtons />
          </div>
          <div className="c-EditoForm__Form">
            <div className="c-EditoForm__Content">
              <EditoStaticFields />
              <EditoDynamicFields
                blockOptions={dynamicFieldsOptions}
                defaultValues={data?.blocks}
              />
            </div>
            <div className="c-EditoForm__SideBar">
              <EditoSideBar status={formData?.status} />
            </div>
          </div>
        </form>
      </FormProvider>
    </>
  );
}
