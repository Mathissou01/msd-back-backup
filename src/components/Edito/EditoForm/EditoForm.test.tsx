import { MockedProvider } from "@apollo/client/testing";
import { render, screen } from "@testing-library/react";
import React from "react";
import { IEditoFields, TDynamicFieldOption } from "../../../lib/edito";
import { valueToEStatus } from "../../../lib/status";
import EditoForm from "./EditoForm";
import { defaultMockData } from "../../../../__mocks__/editoFormMockData";

jest.mock("../../../graphql/client", () => null);

const mocks: {
  dynamicFieldOptions: Array<TDynamicFieldOption>;
  data: IEditoFields;
  formLabels: {
    staticTitle: string;
    staticTags: string;
    staticTagsDescription: string;
    staticImage: string;
    staticImageValidation: string;
    staticImagePlaceholder: string;
    staticShortDescription: string;
    staticShortDescriptionMaxCharacters: string;
  };
} = {
  dynamicFieldOptions: [
    "ComponentBlocksSubHeading",
    "ComponentBlocksHorizontalRule",
  ],
  data: {
    id: "0",
    status: valueToEStatus("draft"),
    title: "mock title",
    shortDescription: "description",
    image: {
      __typename: "UploadFileEntity",
      id: "1",
      attributes: {
        name: "name",
        hash: "hash",
        mime: "mime",
        provider: "provider",
        size: 123,
        url: "url",
      },
    },
    blocks: [
      {
        __typename: "ComponentBlocksSubHeading",
        id: "140",
        subHeadingText: "this is a subHeadingText value",
        subHeadingTag: "h3",
      },
      {
        __typename: "ComponentBlocksHorizontalRule",
        id: "136",
        hr: "<hr>",
      },
    ],
  },
  formLabels: {
    staticTitle: "Titre de l'actualité",
    staticTags: "Thématique",
    staticTagsDescription: "(Tags)",
    staticImage: "Vignette",
    staticImageValidation:
      "Format carré, format .gif, .svg, .png ou .jpg, 30 Mo maximum",
    staticImagePlaceholder:
      "Cliquer pour ajouter une image depuis la bibliothèque de média ou glissez-déposez une image dans cette zone.",
    staticShortDescription: "Description courte",
    staticShortDescriptionMaxCharacters:
      "caractères maximum, affichés dans l'aperçu de l'actualité",
  },
};

describe("EditoForm", () => {
  it("renders", async () => {
    const onSubmit = jest.fn();
    const { container } = render(
      <MockedProvider mocks={defaultMockData}>
        <EditoForm
          data={mocks.data}
          dynamicFieldsOptions={mocks.dynamicFieldOptions}
          onSubmitValid={onSubmit}
          labels={mocks.formLabels}
        />
      </MockedProvider>,
    );

    expect(screen.getByDisplayValue("mock title")).toBeInTheDocument();
    expect(
      screen.getByDisplayValue("this is a subHeadingText value"),
    ).toBeInTheDocument();
    expect(await container).toMatchSnapshot();
  });
});
