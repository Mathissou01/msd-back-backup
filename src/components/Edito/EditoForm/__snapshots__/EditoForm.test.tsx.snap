// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditoForm renders 1`] = `
<div>
  <form
    class="c-EditoForm"
  >
    <div
      class="c-EditoForm__Buttons"
    >
      <div
        class="c-EditoButtons"
      >
        <button
          class="c-CommonButton c-CommonButton_secondary c-CommonButton_fontSmall"
          type="button"
        >
          <div
            class="c-CommonButton__Picto c-CommonButton__Picto_windowUpload"
          />
          Publier
        </button>
        <button
          class="c-CommonButton c-CommonButton_disabled c-CommonButton_primary c-CommonButton_fontSmall"
          disabled=""
          type="submit"
        >
          <div
            class="c-CommonButton__Picto c-CommonButton__Picto_check"
          />
          Enregistrer en tant que brouillon
        </button>
      </div>
    </div>
    <div
      class="c-EditoForm__Form"
    >
      <div
        class="c-EditoForm__Content"
      >
        <div
          class="c-EditoStaticFields"
        >
          <span
            class="c-EditoStaticFields__RequiredLabel"
          >
            Tous les champs marqués d'une * sont obligatoires.
          </span>
          <div
            class="c-FormInput"
          >
            <div
              class="c-FormLabel"
            >
              <label
                class="c-FormLabel__LabelWrapper"
                for="title"
              >
                <span
                  class="c-FormLabel__Label"
                >
                  Titre de l'actualité *
                </span>
              </label>
              <div
                class="c-FormLabel__Content"
              >
                <input
                  aria-errormessage="title_error"
                  aria-invalid="false"
                  class="c-FormInput__Input"
                  data-testid="form-input"
                  id="title"
                  name="title"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            class="c-EditoStaticFields__Thematique"
          >
            <div
              class="c-FormLabel"
            >
              <label
                class="c-FormLabel__LabelWrapper"
              >
                <span
                  class="c-FormLabel__Label"
                >
                  Thématique
                </span>
              </label>
              <div
                class="c-FormLabel__Content"
              >
                <div
                  class="c-CommonSelect"
                >
                  <div
                    class=" css-b62m3t-container"
                  >
                    <span
                      class="css-1f43avz-a11yText-A11yText"
                      id="react-select-2-live-region"
                    />
                    <span
                      aria-atomic="false"
                      aria-live="polite"
                      aria-relevant="additions text"
                      class="css-1f43avz-a11yText-A11yText"
                    />
                    <div
                      class="form-single-multiselect__control css-13cymwt-control"
                    >
                      <div
                        class="form-single-multiselect__value-container form-single-multiselect__value-container--is-multi css-1fdsijx-ValueContainer"
                      >
                        <div
                          class="form-single-multiselect__placeholder css-1jqq78o-placeholder"
                          id="react-select-2-placeholder"
                        />
                        <div
                          class="form-single-multiselect__input-container css-qbdosj-Input"
                          data-value=""
                        >
                          <input
                            aria-autocomplete="list"
                            aria-describedby="react-select-2-placeholder"
                            aria-expanded="false"
                            aria-haspopup="true"
                            autocapitalize="none"
                            autocomplete="off"
                            autocorrect="off"
                            class="form-single-multiselect__input"
                            id="react-select-2-input"
                            role="combobox"
                            spellcheck="false"
                            style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                            tabindex="0"
                            type="text"
                            value=""
                          />
                        </div>
                      </div>
                      <div
                        class="form-single-multiselect__indicators css-1hb7zxy-IndicatorsContainer"
                      >
                        <span
                          class="form-single-multiselect__indicator-separator css-1u9des2-indicatorSeparator"
                        />
                        <div
                          aria-hidden="true"
                          class="form-single-multiselect__indicator form-single-multiselect__dropdown-indicator css-1xc3v61-indicatorContainer"
                        >
                          <svg
                            aria-hidden="true"
                            class="css-tj5bde-Svg"
                            focusable="false"
                            height="20"
                            viewBox="0 0 20 20"
                            width="20"
                          >
                            <path
                              d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                            />
                          </svg>
                        </div>
                      </div>
                    </div>
                    <div>
                      <input
                        name="tags"
                        type="hidden"
                        value=""
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="c-EditoStaticFields__DescriptionInput"
          >
            <div
              class="c-FormInput"
            >
              <div
                class="c-FormLabel"
              >
                <label
                  class="c-FormLabel__LabelWrapper"
                  for="shortDescription"
                >
                  <span
                    class="c-FormLabel__Label"
                  >
                    Description courte *
                  </span>
                  <span
                    class="c-FormLabel__Validation"
                  >
                    80 caractères maximum, affichés dans l'aperçu de l'actualité
                  </span>
                </label>
                <div
                  class="c-FormLabel__Content"
                >
                  <textarea
                    aria-errormessage="shortDescription_error"
                    aria-invalid="false"
                    class="c-FormInput__Input"
                    data-testid="form-input"
                    id="shortDescription"
                    maxlength="80"
                    name="shortDescription"
                    type="text"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <ul
          class="c-EditoDynamicFields"
        >
          <div
            class="c-EditoDynamicFields__Block"
            data-flip-config="{"translate":true,"scale":true,"opacity":true}"
            data-flip-id="ComponentBlocksSubHeading_140"
            data-portal-key="portal"
          >
            <div
              class="c-EditoBlockWrapper"
            >
              <div
                class="c-EditoBlockWrapper__Header"
              >
                <div
                  class="c-EditoBlockWrapper__LeftGroup"
                >
                  <button
                    class="c-EditoBlockWrapper__Action c-EditoBlockWrapper__Action_chevronUp"
                    type="button"
                  />
                  <div
                    class="c-EditoBlockWrapper__Info"
                  >
                    <div
                      class="c-EditoBlockWrapper__Picto c-EditoBlockWrapper__Picto_text"
                    />
                    <div
                      class="c-EditoBlockWrapper__Title"
                    >
                      Titre
                    </div>
                  </div>
                </div>
                <div
                  class="c-EditoBlockWrapper__Actions"
                >
                  <button
                    class="c-EditoBlockWrapper__Action c-EditoBlockWrapper__Action_arrowUp c-EditoBlockWrapper__Action_disabled"
                    disabled=""
                    type="button"
                  />
                  <button
                    class="c-EditoBlockWrapper__Action c-EditoBlockWrapper__Action_arrowDown"
                    type="button"
                  />
                  <button
                    class="c-EditoBlockWrapper__Action c-EditoBlockWrapper__Action_duplicate"
                    type="button"
                  />
                  <button
                    class="c-EditoBlockWrapper__Action c-EditoBlockWrapper__Action_delete"
                    type="button"
                  />
                </div>
              </div>
              <div
                class="c-EditoBlockWrapper__Content c-EditoBlockWrapper__Content_open"
              >
                <div
                  class="c-SubHeadingRow"
                >
                  <div
                    class="c-SubHeadingRow__Description"
                  >
                    <div
                      class="c-FormInput"
                    >
                      <div
                        class="c-FormLabel"
                      >
                        <label
                          class="c-FormLabel__LabelWrapper"
                          for="blocks.0.subHeadingText"
                        >
                          <span
                            class="c-FormLabel__Label"
                          >
                            Texte du titre *
                          </span>
                        </label>
                        <div
                          class="c-FormLabel__Content"
                        >
                          <input
                            aria-errormessage="blocks.0.subHeadingText_error"
                            aria-invalid="false"
                            class="c-FormInput__Input"
                            data-testid="form-input"
                            id="blocks.0.subHeadingText"
                            name="blocks.0.subHeadingText"
                            type="text"
                            value=""
                          />
                        </div>
                      </div>
                    </div>
                  </div>
                  <div>
                    <div
                      class="c-FormSelect"
                    >
                      <div
                        class="c-FormLabel"
                      >
                        <label
                          class="c-FormLabel__LabelWrapper"
                          for="blocks.0.subHeadingTag"
                        >
                          <span
                            class="c-FormLabel__Label"
                          >
                            Niveau de titre *
                          </span>
                        </label>
                      </div>
                      <div
                        class="o-SelectWrapper"
                      >
                        <select
                          aria-errormessage="blocks.0.subHeadingTag_error"
                          aria-invalid="false"
                          class="o-SelectWrapper__Select"
                          data-testid="form-select"
                          id="blocks.0.subHeadingTag"
                        >
                          <option
                            disabled=""
                            hidden=""
                            value="-1"
                          >
                            - Sélectionnez un élément -
                          </option>
                          <option
                            class="o-SelectWrapper__Option"
                            value="0"
                          >
                            H2
                          </option>
                          <option
                            class="o-SelectWrapper__Option"
                            value="1"
                          >
                            H3
                          </option>
                          <option
                            class="o-SelectWrapper__Option"
                            value="2"
                          >
                            H4
                          </option>
                          <option
                            class="o-SelectWrapper__Option"
                            value="3"
                          >
                            H5
                          </option>
                          <option
                            class="o-SelectWrapper__Option"
                            value="4"
                          >
                            H6
                          </option>
                        </select>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="c-EditoDynamicFields__Block"
            data-flip-config="{"translate":true,"scale":true,"opacity":true}"
            data-flip-id="ComponentBlocksHorizontalRule_136"
            data-portal-key="portal"
          >
            <div
              class="c-EditoBlockWrapper"
            >
              <div
                class="c-EditoBlockWrapper__Header"
              >
                <div
                  class="c-EditoBlockWrapper__LeftGroup"
                >
                  <div
                    class="c-EditoBlockWrapper__Info"
                  >
                    <div
                      class="c-EditoBlockWrapper__Picto c-EditoBlockWrapper__Picto_expandVertical"
                    />
                    <div
                      class="c-EditoBlockWrapper__Title"
                    >
                      Séparateur
                    </div>
                  </div>
                </div>
                <div
                  class="c-EditoBlockWrapper__Actions"
                >
                  <button
                    class="c-EditoBlockWrapper__Action c-EditoBlockWrapper__Action_arrowUp"
                    type="button"
                  />
                  <button
                    class="c-EditoBlockWrapper__Action c-EditoBlockWrapper__Action_arrowDown c-EditoBlockWrapper__Action_disabled"
                    disabled=""
                    type="button"
                  />
                  <button
                    class="c-EditoBlockWrapper__Action c-EditoBlockWrapper__Action_duplicate"
                    type="button"
                  />
                  <button
                    class="c-EditoBlockWrapper__Action c-EditoBlockWrapper__Action_delete"
                    type="button"
                  />
                </div>
              </div>
              <div
                class="c-EditoBlockWrapper__Content c-EditoBlockWrapper__Content_open c-EditoBlockWrapper__Content_empty"
              >
                <input
                  name="blocks.1.hr"
                  type="hidden"
                  value="<hr>"
                />
              </div>
            </div>
          </div>
        </ul>
        <div
          class="c-EditoDynamicFields__Options"
        >
          <span
            class="c-EditoDynamicFields__OptionsTitle"
          >
            Ajouter un bloc
          </span>
          <div
            class="c-EditoDynamicFields__OptionsList"
          >
            <button
              class="c-CommonButton c-CommonButton_secondary c-CommonButton_fontSmall"
              type="button"
            >
              <div
                class="c-CommonButton__Picto c-CommonButton__Picto_text"
              />
              Titre
            </button>
            <button
              class="c-CommonButton c-CommonButton_secondary c-CommonButton_fontSmall"
              type="button"
            >
              <div
                class="c-CommonButton__Picto c-CommonButton__Picto_expandVertical"
              />
              Séparateur
            </button>
          </div>
        </div>
      </div>
      <div
        class="c-EditoForm__SideBar"
      >
        <div
          class="c-EditoSideBar"
        >
          <div
            class="c-EditoSideBar__Status c-EditoSideBar__Status_draft"
          >
            Brouillon
          </div>
          <div
            class="c-EditoSideBar__UnpublishedDate"
          >
            Date de dépublication
            <div
              class="c-EditoSideBar__UnpublishedDate_datePicker"
            >
              <div
                class="sz-datepicker row"
              >
                <div
                  class="react-date-picker react-date-picker--closed react-date-picker--enabled sz-datepicker__single"
                >
                  <div
                    class="react-date-picker__wrapper"
                  >
                    <div
                      class="react-date-picker__inputGroup"
                    >
                      <input
                        hidden=""
                        max="2099-12-31"
                        min="2023-03-15"
                        name="date"
                        style="visibility: hidden; position: absolute; z-index: -999;"
                        type="date"
                        value=""
                      />
                      <input
                        autocomplete="off"
                        class="react-date-picker__inputGroup__input react-date-picker__inputGroup__day"
                        data-input="true"
                        inputmode="numeric"
                        max="31"
                        min="1"
                        name="day"
                        placeholder="--"
                        type="number"
                        value=""
                      />
                      <span
                        class="react-date-picker__inputGroup__divider"
                      >
                        /
                      </span>
                      <input
                        autocomplete="off"
                        class="react-date-picker__inputGroup__input react-date-picker__inputGroup__month"
                        data-input="true"
                        inputmode="numeric"
                        max="12"
                        min="1"
                        name="month"
                        placeholder="--"
                        type="number"
                        value=""
                      />
                      <span
                        class="react-date-picker__inputGroup__divider"
                      >
                        /
                      </span>
                      <input
                        autocomplete="off"
                        class="react-date-picker__inputGroup__input react-date-picker__inputGroup__year"
                        data-input="true"
                        inputmode="numeric"
                        max="2099"
                        min="2023"
                        name="year"
                        placeholder="----"
                        step="1"
                        type="number"
                        value=""
                      />
                    </div>
                    <button
                      class="react-date-picker__calendar-button react-date-picker__button"
                      type="button"
                    >
                      <i
                        class="sz-icon-line calendar-3 mr-2"
                      />
                      <i
                        class="sz-icon-bold arrow-down-1 small pt-2"
                      />
                    </button>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="c-EditoSideBar__UnpublishedDate_text"
            >
              A la date de dépublication, l’actualité est masquée sur le site mais conservée dans le back-office à l’état archivé.
            </div>
          </div>
          <div
            class="c-DateBlock"
          >
            <div
              class="c-DateBlock__CreationDate"
            >
              <div
                class="c-DateBlock__CreationDateLabel"
              >
                Création
              </div>
              <div
                class="c-DateBlock__CreationDateDateAndTime"
              >
                <div />
              </div>
            </div>
            <div
              class="c-DateBlock__UpdateDate"
            >
              <div
                class="c-DateBlock__UpdateDateLabel"
              >
                Dernière modification
              </div>
              <div
                class="c-DateBlock__UpdateDateDateAndTime"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </form>
</div>
`;
